src/rvsdg/from_egglog.rs

```diff
@@ -24,7 +24,25 @@ impl<'a> RvsdgFromEgglog<'a> {
                 ("Arg", [Term::Lit(Int(n))]) => Operand::Arg(*n as usize),
                 ("Node", [body]) => Operand::Id(self.egglog_term_to_body(body.clone())),
                 ("Project", [Term::Lit(Int(n)), body]) => {
-                    Operand::Project(*n as usize, self.egglog_term_to_body(body.clone()))
+                    let Term::App(bfunc, bargs) = body else {
+                        panic!("expected a body after project, got {}", self.termdag.to_string(body))
+                    };
+                    let bargs = bargs
+                        .iter()
+                        .map(|t| self.termdag.get(*t))
+                        .collect::<Vec<_>>();
+
+                    // Every `OperandGroup` should be used by a
```

Comment:
Could you actually drop a print edge here and fail to evaluate the print because we didn't evaluate the whole Operand Group?
Probably needs to be reflected in the rust RVSDG after all

Reply:
@yihozhang says this is fine because we always extract the state edge at the end.